### ISSUE TITLE
[css-values] Font-relative length definitions don't account for media query usage

### ISSUE THREAD

**Issue by cvrebert** [w3c/csswg-drafts#153]
(All emphases added)

> [...] the [font-relative lengths](https://drafts.csswg.org/css-values/#font-relative-lengths) refer to the font metrics of the **element on which they are used**.
> [...]
> <dl><dt><a href="https://drafts.csswg.org/css-values/#em">em unit</a></dt>
> <dd>Equal to the computed value of the 'font-size' property of the <b>element on which it is used</b>.</dd></dl>

When using such font-relative lengths in a media query expression:
* The author isn't "using" the length "on" something, per se.
* In the most common case, the author is using the length in relation to the viewport, which isn't an *element*.

> <dl><dt><a href="https://drafts.csswg.org/css-values/#rem">rem unit<a></dt>
> <dd>Equal to the <b>computed value</b> of font-size on the root element. When specified on the 'font-size' property [...]</dd></dl>

Media queries interpret `rem` differently; see https://drafts.csswg.org/mediaqueries/#units

The other font-relative unit definitions likewise don't seem to account for usage in media query expressions.

**Comment by frivoal** [comment]
The media query spec does have some generic phrasing:

> Relative length units in media queries are based on the initial value, which means that units are never based on results of declarations. For example, in HTML, the em unit is relative to the initial value of font-size, defined by the user agent or the user’s preferences, not any styling on the page.

This seems sufficient, even for the units whose value normally depend on which element you compute them against, since properties at their initial value will give the same result on any element. We could add something about the root element, as that wouldn't be wrong, and might be clearer to some, but I don't think this would actually make any normative difference, so I am not sure it's a good idea.

As to whether the values and units spec should also say something, or whether it can rely on MQs dealing with it with the above phrasing, I have no strong opinion.

**Comment by tabatkins** [comment]
V&U defines common value types, generally focusing on properties. MQ is a special-case and takes care of that with its own language.  I don't think we need to call anything out in V&U specially for this.

**Comment by frivoal** [comment]
Right. So unless you think a clarification is needed in Media Queries, I think that normatively we're fine. We could still have a note in V&U giving a hint that in the MQ context, the interpretation is special (with a link to MQ), though.
